sphinx/ext/coverage.py

```diff
@@ -167,11 +208,43 @@ def ignore_pyobj(self, full_name: str) -> bool:
         )
 
     def build_py_coverage(self) -> None:
-        objects = self.env.domaindata['py']['objects']
-        modules = self.env.domaindata['py']['modules']
+        seen_objects = self.env.domaindata['py']['objects']
+        seen_modules = self.env.domaindata['py']['modules']
 
         skip_undoc = self.config.coverage_skip_undoc_in_source
 
+        # Figure out which of the two operating modes to use:
+        #
+        # - If 'coverage_modules' is not specified, we check coverage for all modules
+        #   seen in the documentation tree. Any objects found in these modules that are
+        #   not documented will be noted. This will therefore only identify missing
+        #   objects but it requires no additional configuration.
+        # - If 'coverage_modules' is specified, we check coverage for all modules
+        #   specified in this configuration value. Any objects found in these modules
+        #   that are not documented will be noted. In addition, any objects from other
+        #   modules that are documented will be noted. This will therefore identify both
+        #   missing modules and missing objects but it requires manual configuration.
+        if not self.modules:
+            modules = set(seen_modules)
+        else:
+            modules = set()
+            for mod_name in self.modules:
+                try:
+                    modules |= load_modules(mod_name, self.mod_ignorexps)
+                except ImportError as err:
+                    logger.warning(__('module %s could not be imported: %s'), mod_name, err)
```

Comment:
Since people like to suppress warnings, could you also decide for a type/subtype for the log record and add it to the existing ones.

Reply:
My one disagreement: none of the other `logger.warning` calls in here do this, so I think we want to do it for all or none. Can I do this as a follow-up? I've added a TODO.

Reply:
Sure! Overall, we need to add those type/subtype more or less everywhere where warnings could be annoying because people request the ability to suppress them (and to turn them as non-errors), so you could have a separate issue for warnings regarding coverage-related things.